include/aunit/reporters/aunit-reporter-junit.adb

```diff
@@ -0,0 +1,150 @@
+-- GNAT is free software;  you can  redistribute it  and/or modify it under --
+-- terms of the  GNU General Public License as published  by the Free Soft- --
+-- ware  Foundation;  either version 3,  or (at your option) any later ver- --
+-- sion.  GNAT is distributed in the hope that it will be useful, but WITH- --
+-- OUT ANY WARRANTY;  without even the  implied warranty of MERCHANTABILITY --
+-- or FITNESS FOR A PARTICULAR PURPOSE.                                     --
+--                                                                          --
+-- As a special exception under Section 7 of GPL version 3, you are granted --
+-- additional permissions described in the GCC Runtime Library Exception,   --
+-- version 3.1, as published by the Free Software Foundation.               --
+--                                                                          --
+-- You should have received a copy of the GNU General Public License and    --
+-- a copy of the GCC Runtime Library Exception along with this program;     --
+-- see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see    --
+-- <http://www.gnu.org/licenses/>.                                          --
+--                                                                          --
+-- GNAT is maintained by AdaCore (http://www.adacore.com)                   --
+--                                                                          --
+------------------------------------------------------------------------------
+
+with AUnit.IO;              use AUnit.IO;
+with AUnit.Time_Measure;    use AUnit.Time_Measure;
+
+package body AUnit.Reporter.JUnit is
+   
+   procedure Put_Special_Chars (File : File_Type; S : String) is
+   begin
+      for C of S loop
+         case C is
+            when '"' => Put (File, "&quot;");
+            when '<' => Put (File, "&lt;");
+            when '>' => Put (File, "&gt;");
+            when '&' => Put (File, "&amp;");
+            when others => Put (File, (1 => C));
+         end case;
+      end loop;
+   end Put_Special_Chars;
+   
+   procedure Put_Measure is
+     new Gen_Put_Measure_In_Seconds;
+   
+   procedure Report_Test (File : File_Type; Test : Test_Result) is
+   begin
+      Put (File, "<testcase name=""");
+      Put_Special_Chars (File, Test.Test_Name.all &
+           (if Test.Routine_Name = null then ""
+              else " : " & Test.Routine_Name.all));
+      Put (File, """ time=""");
+      Put_Measure (File, Get_Measure (Test.Elapsed));
+      if Test.Failure /= null then
+         Put_Line (File, """>"); 
+         Put_Line (File, "<failure>");
+         Put_Line (File, "<![CDATA[");
+         Put (File, "        Assertion: ");
+         Put_Line (File, Test.Failure.Message.all);
+         Put (File, "        File: ");
+         Put_Line (File, Test.Failure.Source_Name.all);
+         Put (File, "        Line: ");
+         Put (File, Integer (Test.Failure.Line));
```

Comment:
With recent addition of file_io on new runtimes Put for integer now has a padding by default, so the output looks like:
```
        Assertion: Test not implemented.
        File: some_unit-test_data-tests.adb
        Line:          76
```
I suspect this is undesirable? Adding 0 as third parameter of Put for Integer makes it look like this:
```
        Assertion: Test not implemented.
        File: some_unit-test_data-tests.adb
        Line: 76
```